spec/formatter_spec.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const {format} = require('../src/fxp');
+
+describe('XML formatter', () => {
+  it('should format simple document', () => {
+    const result = format('<data><item>1</item></data>');
+    expect(result).toBe(`
+<data>
+  <item>
+    1
+  </item>
+</data>`);
+  });
+
+  it('should preserve XML PI tag', () => {
+    const result = format('<?xml version="1.0"?><data>hello</data>');
+    expect(result).toBe(`
+<?xml   version="1.0"?>
+<data>
+  hello
+</data>`);
+  });
+
+  it('should preserve self-closed tags', () => {
+    const result = format('<data><self-closed/></data>');
+    expect(result).toBe(`
+<data>
+  <self-closed/>
+</data>`);
+  });
+
+  it('should preserve comments', () => {
+    const result = format('<data><!-- comment --><value>1</value></data>');
+    expect(result).toBe(`
+<data>
+  <!-- comment -->
+  <value>
+    1
+  </value>
+</data>`);
+  });
+
+  it('should preserve attributes', () => {
+    const result = format('<data kind="text">hello</data>');
+    expect(result).toBe(`
+<data kind="text">
+  hello
+</data>`);
+  });
+
+  xit('should preserve closing tag after empty content', () => {
+    const result = format('<empty></empty>');
+    expect(result).toBe(`
+<empty>
+</empty>`);
```

Comment:
I could not find a way how to make this test pass, while keeping `<self-closed/>` test to pass too. When `suppressEmptyNode` is enabled, `<self-closed/>` passes `<empty></empty>` fails. When I disable that setting, then the first test fails and this one passes.

If we want to support both cases, then I think we need to improve the parser to store metadata allowing the builder to distinguish between a self-closed element and an element with empty content.

WDYT?